src/nodeinfoprovider/nodeinfoprovider.cpp

```diff
@@ -132,7 +132,13 @@ Error NodeInfoProvider::UnsubscribeNodeStatusChanged(iam::nodeinfoprovider::Node
 
 Error NodeInfoProvider::InitNodeID()
 {
-#ifndef CONFIG_NATIVE_APPLICATION
+#ifdef CONFIG_NATIVE_APPLICATION
+    if (auto err = FS::ReadFileToString(cNodeIDFile, mNodeInfo.mNodeID); !err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    mNodeInfo.mNodeID.Trim("\r\n");
+#else
     char buffer[cNodeIDLen + 1] {};
 
     auto ret = hwinfo_get_device_id(reinterpret_cast<uint8_t*>(buffer), sizeof(buffer) - 1);
```

Comment:
Probably should be moved to FS::ReadFileToString ?

Reply:
It is not required after your PR.